presto-vertica/src/main/java/io/prestosql/plugin/vertica/VerticaClient.java

```diff
@@ -0,0 +1,223 @@
+    {
+        Optional<ColumnMapping> mappingToVarchar = getForcedMappingToVarchar(typeHandle);
+        if (mappingToVarchar.isPresent()) {
+            return mappingToVarchar;
+        }
+
+        switch (typeHandle.getJdbcType()) {
+            case Types.BIT:
+                return Optional.of(booleanColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.DOUBLE:
+            case Types.FLOAT:
+                return Optional.of(doubleColumnMapping());
+            case Types.NUMERIC:
+                int decimalDigits = typeHandle.getRequiredDecimalDigits();
+                int precision = typeHandle.getRequiredColumnSize() + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
+                if (precision > Decimals.MAX_PRECISION) {
+                    return Optional.empty();
+                }
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(createCharType(typeHandle.getRequiredColumnSize())));
+            case Types.VARCHAR:
+                return Optional.of(varcharColumnMapping(createVarcharType(typeHandle.getRequiredColumnSize())));
+            case Types.DATE:
+                return Optional.of(dateColumnMapping());
+        }
+        if (getUnsupportedTypeHandling(session) == CONVERT_TO_VARCHAR) {
+            return mapToUnboundedVarchar(typeHandle);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "varchar";
+            }
+            else {
+                dataType = "varchar(" + varcharType.getBoundedLength() + ")";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
+        }
+        if (type instanceof CharType) {
+            return WriteMapping.sliceMapping("char(" + ((CharType) type).getLength() + ")", charWriteFunction());
+        }
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.sliceMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+
+        WriteMapping writeMapping = WRITE_MAPPINGS.get(type);
+        if (writeMapping != null) {
+            return writeMapping;
+        }
```

Comment:
I find it more readable not to use a map for this. 
Also, it;s nice to order the mappings in some specific order.
My preferred ordering is:

- boolean
- integer types from smallest to biggest
- real, double
- decimal
- char, varchar, varbinary
- date
- time, time tz
- timestamp, timestamp tz